### PR TITLE
[Enhancement](MaxCompute)Refactoring maxCompute catalog using Storage API.

### PR DESCRIPTION
**Pull request by hubgeter**
## Proposed changes
Refactoring maxCompute catalog using Storage API.

Storage API : https://help.aliyun.com/zh/maxcompute/user-guide/open-storage-sample-java-sdk?spm=a2c4g.11186623.0.i0

```
The following are required:

CREATE CATALOG mc PROPERTIES (
"type" = "max_compute",
"mc.default.project" = "xxx",
"mc.access_key" = "xxx",
"mc.secret_key" = "xxxx",
"mc.endpoint" = "xxxx"
);

Optional parameters:
Configuration Item      Default Value
"mc.quota"              =    "pay-as-you-go"

"mc.split_strategy" =   "byte_size" Split according to file size
"mc.split_byte_size" = "268435456" You can set the file size of each split

"mc.split_strategy" = "row_count" Split according to the number of rows of data
"mc.split_row_count" = "1048576" You can set how many lines to read for each split
```

### CODE REVIEW

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

File: regression-test/suites/external_table_p2/maxcompute/test_max_compute_complex_type.groovy [hunk 1]
```diff
@@ -142,7 +151,7 @@
         )
     );
  */
-suite("test_max_compute_complex_type", "p0,external,doris,external_docker,external_docker_doris") {
+suite("test_max_compute_complex_type", "p2,external,doris,external_docker,external_docker_doris") {
```
Comment (morningman) [inline regression-test/suites/external_table_p2/maxcompute/test_max_compute_complex_type.groovy:154]: p2,external,maxcompute,external_remote,external_remote_maxcompute

File: regression-test/suites/external_table_p2/maxcompute/test_external_catalog_maxcompute.groovy [hunk 2]
```diff
@@ -14,8 +14,273 @@
...
+      'County',
+      'WA',
+      '45678',
+      'USA',
+      CAST(-7.0 AS DOUBLE),
+      CAST(8.00 AS DECIMAL(5,2))
+    );
+  
+ */
+suite("test_external_catalog_maxcompute", "p2,external,doris,external_docker,external_docker_doris") {
```
Comment (morningman) [inline regression-test/suites/external_table_p2/maxcompute/test_external_catalog_maxcompute.groovy:283]: p2,external,maxcompute,external_remote,external_remote_maxcompute

File: regression-test/suites/external_table_p2/maxcompute/test_max_compute_all_type.groovy [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+  ARRAY(null, 'value1', 'value2'), -- t_array_string_starting_with_nulls
+  ARRAY('value1', null, 'value2'), -- t_array_string_with_nulls_in_between
+  ARRAY('value1', 'value2', null) -- t_array_string_ending_with_nulls
+);
+
+ALTER TABLE mc_all_types MERGE SMALLFILES;
+
+select * from mc_all_types;
+ */
+suite("test_max_compute_all_type", "p2,external,doris,external_docker,external_docker_doris") {
```
Comment (morningman) [inline regression-test/suites/external_table_p2/maxcompute/test_max_compute_all_type.groovy:315]: p2,external,maxcompute,external_remote,external_remote_maxcompute

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java [hunk 4]
```diff
@@ -75,11 +99,290 @@ private void setScanParams(TFileRangeDesc rangeDesc, MaxComputeSplit maxComputeS
...
+    }
+
+    private String convertLiteralToOdpsValues(OdpsType odpsType, Expr expr) throws AnalysisException {
+        if (!(expr instanceof LiteralExpr)) {
+            throw new AnalysisException("Do not support convert ["
+                    + expr.getExprName() + "] in convertSlotRefToAttribute.");
+        }
+        LiteralExpr literalExpr = (LiteralExpr) expr;
+
+        literalExpr.toString();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java:348]: why call this here?

**Comment by hubgeter** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**